passes/sat/qbfsat.cc

```diff
@@ -98,11 +98,8 @@ dict<std::string, std::string> get_hole_loc_name_map(RTLIL::Module *module, cons
 	for (auto cell : module->cells()) {
 		std::string cell_src = cell->get_src_attribute();
 		auto pos = sol.hole_to_value.find(cell_src);
-		if (pos != sol.hole_to_value.end()) {
-#ifndef NDEBUG
-			log_assert(cell->type.in("$anyconst", "$anyseq"));
-			log_assert(cell->getPort(ID::Y).is_wire());
-#endif
+		if (pos != sol.hole_to_value.end() && cell->type.in("$anyconst", "$anyseq")) {
+			log_assert(hole_loc_to_name.find(pos->first) == hole_loc_to_name.end());
 			hole_loc_to_name[pos->first] = cell->getPort(ID::Y).as_wire()->name.str();
```

Comment:
Wait. Doesn't this break if more than one `$anyconst` appears on a single line?

Reply:
It breaks if more than one `$anyconst` has the exact same source location, including `first_column`, `first_line`, `last_column`, and `last_line`.  But that can only happen if the location is e.g. copied multiple times from a single cell within Yosys (some user pass?), and it can never happen organically from reading the file.  Please also see 75ce3ba4992ce1c4376c7e258395a113f0ae94ad for more updated handling of `$anyconst` source locations.